src/main/java/bio/terra/service/journal/JournalDao.java

```diff
@@ -0,0 +1,162 @@
+package bio.terra.service.journal;
+
+import bio.terra.model.JournalEntryModel;
+import bio.terra.service.auth.iam.IamResourceType;
+import bio.terra.service.snapshot.exception.CorruptMetadataException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import javax.validation.constraints.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+@Repository
+public class JournalDao {
+
+  public static final String TABLE_NAME = "journal";
+  private static final Logger logger = LoggerFactory.getLogger(JournalDao.class);
+  private final NamedParameterJdbcTemplate jdbcTemplate;
+
+  private final ObjectMapper objectMapper;
+
+  private static final String summaryQueryColumns =
+      "id, entry_type, user_email, resource_key, resource_type, "
+          + "class_name, method_name, mutation, note, entry_timestamp ";
+
+  @Autowired
+  public JournalDao(NamedParameterJdbcTemplate jdbcTemplate, ObjectMapper objectMapper) {
+    this.jdbcTemplate = jdbcTemplate;
+    this.objectMapper = objectMapper;
+  }
+
+  @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
+  public void create(
+      @NotNull JournalService.EntryType entryType,
+      @NotNull String user,
+      @NotNull UUID resource_key,
+      @NotNull IamResourceType resourceType,
+      @NotNull String className,
+      @NotNull String methodName,
+      String note,
+      String mutationAsJson) {
+    String sql =
+        "INSERT INTO "
+            + TABLE_NAME
+            + " "
+            + "(entry_type, user_email, resource_key, resource_type, class_name, method_name, "
+            + "mutation, note, entry_timestamp) "
+            + "VALUES(:entryType, :user, :resource_key, :resourceType, :className, :methodName, "
+            + "COALESCE(cast(:mutation as jsonb)), COALESCE(:note), current_timestamp)";
```

Comment:
I'm not sure what the coalesce with one value does..., does they need default values?

Reply:
They don't need default values, but my understanding was that if they "COALESCE" to null and the fields are optional, a value won't actually be inserted for the *optional* field.  It's likely unnecessary.  Happy to remove them if you like me to.